lib/bamboo/phoenix.ex

```diff
@@ -1,287 +1,289 @@
-defmodule Bamboo.Phoenix do
-  @moduledoc """
-  Render templates and layouts with Phoenix.
-
-  This module makes it very easy to render layouts and views using Phoenix.
-  Pass an atom (e.g. `:welcome_email`) as the template name to render HTML and
-  plain text emails. Use a string if you only want to render one type, e.g.
-  `"welcome_email.text"` or `"welcome_email.html"`.
-
-  ## Examples
-
-      defmodule Email do
-        use Bamboo.Phoenix, view: MyApp.EmailView
-
-        def text_and_html_email_with_layout do
-          new_email()
-          # You could set just a text layout or just an html layout
-          |> put_text_layout({MyApp.LayoutView, "email.text"})
-          |> put_html_layout({MyApp.LayoutView, "email.html"})
-          # Or you can set a layout for both html and text at the same time
-          |> put_layout({MyApp.LayoutView, :email})
-          # Pass an atom to render html AND plain text templates
-          |> render(:text_and_html_email)
-        end
-
-        def text_and_html_email_without_layouts do
-          new_email()
-          |> render(:text_and_html_email)
-        end
-
-        def email_with_assigns(user) do
-          new_email()
-          # @user will be available in the template
-          |> render(:email_with_assigns, user: user)
+if Code.ensure_loaded?(Phoenix) do
```

Comment:
BTW, there is another way to do it, removing `only: :test` from phoenix dep declaration. But I found this one a little bit better since bamboo itself does not depend on phoenix.

Going far away in my imagination, I guess it would be a good idea to separate some of Phoenix and Plug code from here. This could be an umbrella project with three applications: `bamboo` with the basics, `bamboo_phoenix` to use some phoenix views to render and `bambo_viewer` for the email viewer. 

Will be happy to open a PR to it if you liked the umbrella idea.